src/checkout.c

```diff
@@ -1527,18 +1547,26 @@ static int blob_content_to_file(
 
 	if ((fd = p_open(path, flags, mode)) < 0) {
 		giterr_set(GITERR_OS, "could not open '%s' for writing", path);
-		return fd;
+		error = fd;
+		goto cleanup;
 	}
 
 	filter_opts.attr_session = &data->attr_session;
-	filter_opts.temp_buf = &data->tmp;
+	filter_opts.temp_buf = &temp_buf;
```

Comment:
I wonder if that is why I needed to build a mutex around the filters?

Reply:
I… think it does, yes. I've singled that out because there are some other users of that `data->tmp` buffer and thought they might clobber as well, but after more time reading the code, it seems to only serve as a temporary buffer for the filter application (ie. it shouldn' t share its lifetime). So it makes sense IMHO to just use a stack-allocated buf at the right level.

The need for locking is still unconfirmed though 😉.